instructure/canvas-android
# [MBL-15978][Student][Teacher] Dark Mode

Test notes:
- Webviews and Flutter features (everything on the calendar screen) will be implemented later, disregard those problems for now.
- There were some changes to the light mode UI as well some of them you can find in the detailed commit messages. (Mostly removal of some unnecessary divider, some slight color changes and UI improvements to add touch feedback to items).
- Maybe some brand colors and course colors will look bad in dark mode. Don't need to note this as a bug, we will discuss later what to do with these.
- Verify that every feature and every screen works correctly in dark mode, and all the text is correctly visible. You can use the detailed smoke test plan from Confluence.
- Smoke test the light mode UI. This does not need to be as detailed.
- Verify switching themes both in the app and in the device settings when the app is set to system default.

Review notes:
- There are lots of changes in this PR, the best way to review this is by commits. All the commits have detailed messages about what changed and it helps to understand the underlying changes more quickly.

refs: MBL-15978
affects: Teacher, Student
release note: Added support for dark mode.
